Index: BaseJspScreen.java
===================================================================
RCS file: /home/cvspublic/jakarta-turbine-2/src/java/org/apache/turbine/modules/screens/BaseJspScreen.java,v
retrieving revision 1.1.1.1
diff -u -r1.1.1.1 BaseJspScreen.java
--- BaseJspScreen.java	16 Aug 2001 05:08:37 -0000	1.1.1.1
+++ BaseJspScreen.java	11 Feb 2003 14:14:32 -0000
@@ -82,9 +82,6 @@
     public ConcreteElement buildTemplate( RunData data )
         throws Exception
     {
-        // set up any data in beans, etc
-        doBuildTemplate(data);
-
         // get the name of the JSP we want to use
         String templateName = TurbineTemplate.getScreenTemplateName(
             data.getTemplateInfo().getScreenTemplate() );
